ebean-core/src/main/java/io/ebeaninternal/server/query/SqlTreeNodeRoot.java

```diff
@@ -76,6 +76,7 @@ public SqlJoinType appendFromBaseTable(DbSqlContext ctx, SqlJoinType joinType) {
 
   @Override
   public void dependentTables(Set<String> tables) {
+    tables.add(baseTable);
```

Comment:
This is effectively the bug fix here. We need the baseTable to be included in the set of dependent tables